File: sdks/jvm/src/main/java/com/vsnexa/v6/NexaSdkException.java
package com.vsnexa.v6;

public final class NexaSdkException extends RuntimeException {
  private final String code;
  private final String serverCode;
  private final Object details;

  public NexaSdkException(String code) { this(code, null, null); }
  public NexaSdkException(String code, Object details) { this(code, null, details); }
  public NexaSdkException(String code, String serverCode, Object details) {
    super(code);
    this.code = code;
    this.serverCode = serverCode;
    this.details = details;
  }
  public String code() { return code; }
  public String serverCode() { return serverCode; }
  public Object details() { return details; }
}

File: sdks/dotnet/src/Nexa.V6.Sdk/NexaSdkException.cs
namespace Nexa.V6.Sdk;

public sealed class NexaSdkException : Exception
{
    public string Code { get; }
    public string? ServerCode { get; }
    public object? Details { get; }

    public NexaSdkException(string code, object? details = null, string? serverCode = null)
        : base(code)
    {
        Code = code;
        Details = details;
        ServerCode = serverCode;
    }
}
